server/src/main/java/io/unitycatalog/server/service/ModelService.java

```diff
@@ -3,102 +3,260 @@
       @Param("catalog_name") Optional<String> catalogName,
       @Param("schema_name") Optional<String> schemaName,
       @Param("max_results") Optional<Integer> maxResults,
       @Param("page_token") Optional<String> pageToken) {
-    return HttpResponse.ofJson(
-        MODEL_REPOSITORY.listRegisteredModels(catalogName, schemaName, maxResults, pageToken));
+
+    ListRegisteredModelsResponse listRegisteredModelsResponse =
+            MODEL_REPOSITORY.listRegisteredModels(catalogName, schemaName, maxResults, pageToken);
+    filterModels("""
+            #authorize(#principal, #metastore, OWNER) ||
+            #authorize(#principal, #registered_model, OWNER) ||
+            (#authorize(#principal, #registered_model, EXECUTE) && #authorize(#principal, #schema, USE_SCHEMA) && #authorize(#principal, #catalog, USE_CATALOG))
+            """, listRegisteredModelsResponse.getRegisteredModels());
+
+    return HttpResponse.ofJson(listRegisteredModelsResponse);
   }
 
   @Get("/{full_name}")
-  public HttpResponse getRegisteredModel(@Param("full_name") String fullNameArg) {
+  @AuthorizeExpression("""
+          #authorize(#principal, #metastore, OWNER) ||
+          #authorize(#principal, #registered_model, OWNER) ||
+          (#authorize(#principal, #registered_model, EXECUTE) && #authorize(#principal, #schema, USE_SCHEMA) && #authorize(#principal, #catalog, USE_CATALOG))
+          """)
+  @AuthorizeKey(METASTORE)
+  public HttpResponse getRegisteredModel(@Param("full_name") @AuthorizeKey(REGISTERED_MODEL) String fullNameArg) {
     assert fullNameArg != null;
     RegisteredModelInfo registeredModelInfo = MODEL_REPOSITORY.getRegisteredModel(fullNameArg);
     return HttpResponse.ofJson(registeredModelInfo);
   }
 
   @Patch("/{full_name}")
-  public HttpResponse updateRegisteredModel(UpdateRegisteredModel updateRegisteredModel) {
+  @AuthorizeExpression("""
+          #authorize(#principal, #metastore, OWNER) ||
+          #authorize(#principal, #catalog, OWNER) ||
+          (#authorize(#principal, #catalog, USE_CATALOG) && #authorize(#principal, #schema, OWNER)) ||
+          (#authorize(#principal, #registered_model, OWNER) && #authorize(#principal, #schema, USE_SCHEMA) && #authorize(#principal, #catalog, USE_CATALOG))
+          """)
+  @AuthorizeKey(METASTORE)
+  public HttpResponse updateRegisteredModel(@Param("full_name") @AuthorizeKey(REGISTERED_MODEL) String fullNameArg, UpdateRegisteredModel updateRegisteredModel) {
     assert updateRegisteredModel != null;
     RegisteredModelInfo updateRegisteredModelResponse =
         MODEL_REPOSITORY.updateRegisteredModel(updateRegisteredModel);
     return HttpResponse.ofJson(updateRegisteredModelResponse);
   }
 
   @Delete("/{full_name}")
+  @AuthorizeExpression("""
+          #authorize(#principal, #metastore, OWNER) ||
+          (#authorize(#principal, #registered_model, OWNER) && #authorize(#principal, #schema, USE_SCHEMA) && #authorize(#principal, #catalog, USE_CATALOG))
```

Comment:
Same question here.  Catalog owner and schema owner would be able to delete the models as well.

Reply:
I added explicit checks for catalog and schema OWNER